development/configuration/configuring-prestashop.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Configuring PrestaShop
+---
+
+# Introduction to configuration
+
+## Storage & access
+
+Most of PrestaShop's configuration is stored in the `ps_configuration` database table and is accessible through the [Configuration storage service][configuration-storage].
+
+Some settings are hard coded as constants.
+
+## Configuration files
+
+PrestaShop has four main configuration files:
+
+* `/config/config.inc.php`
+* `/config/defines.inc.php`
+* `/config/smarty.config.inc.php`
+* `/app/config/parameters.php`
+
+### config.inc.php
+
+It is the main configuration file for PrestaShop. You should not need to touch anything in there.
+
+### defines.inc.php
+
+This file contains PrestaShop constant values.
+
+It also contains the location of all the files and folders. If you need to change their location, do not forget to keep the original path nearby, for instance in a PHP comment, in case you need to revert back to it later on.
+
+You can override constant values from this file by setting them in `/config/defines_custom.inc.php`. This is also a great place to put your custom constants which needs to be available globally in the system.
+
+### smarty.config.inc.php
+
+This file contains all the Smarty-related settings.
+
+The Smarty cache system should always be disabled, as it is not compatible with PrestaShop: keep `$smarty->caching = false;` as it is.
+
+`$smarty->compile_check` should be left to `false` in development mode.
+
+`$smarty->debugging` gives access to Smarty debug information when displaying a page. That setting is more easily modified in the “Performance” page of the advanced parameters menu : the “Debug console” option enables you to choose between never displaying Smarty’s debug information, always displaying it, or only displaying it when you add `?SMARTY_DEBUG` to the URL of the page you want to test, which can be very useful.
```

Comment:
```suggestion
`$smarty->debugging` gives access to Smarty debug information when displaying a page. That setting is more easily modified in the “Performance” page of the advanced parameters menu: the “Debug console” option enables you to choose between never displaying Smarty’s debug information, always displaying it, or only displaying it when you add `?SMARTY_DEBUG` to the URL of the page you want to test, which can be very useful.
```

Reply:
There's no such option in the back office. There's some obsolete code around this, tho 🤔

Reply:
Yep I didn't get Smarty debugging to work. That might be an actual issue though.